inc/components/components/post-list/component.php

```diff
@@ -57,7 +57,9 @@
 			}
 
 			$post_ids = wp_list_pluck( $query->posts, 'ID' );
-			$post_ids = post_list_get_and_add_used_post_ids( $post_ids );
+
+			// Add the current $post_ids to the list of used ids.
+			post_list_get_and_add_used_post_ids( $post_ids );
```

Comment:
Ah! I hadn't noticed how this feature worked. This could cause lots of confusion since it persists throughout the entire page. Separating adding and getting used post IDs might be useful, along with making this more general use outside just post-lists so an individual post or component could add their ID to the queue. No need to do anything here, I'll open a new ticket.

Reply:
You may be interested in this, https://github.com/alleyinteractive/alleypack/tree/master/modules/unique-wp-query/1.0

Reply:
Also, agreed, it could be done better.